tools/tpm2_policycountertimer.c

```diff
@@ -0,0 +1,198 @@
+        } else if (!strcmp(argv[0], "slt")) {
+            ctx.operation = TPM2_EO_SIGNED_LT;
+        } else if (!strcmp(argv[0], "ult")) {
+            ctx.operation = TPM2_EO_UNSIGNED_LT;
+        } else if (!strcmp(argv[0], "sge")) {
+            ctx.operation = TPM2_EO_SIGNED_GE;
+        } else if (!strcmp(argv[0], "uge")) {
+            ctx.operation = TPM2_EO_UNSIGNED_GE;
+        } else if (!strcmp(argv[0], "sle")) {
+            ctx.operation = TPM2_EO_SIGNED_LE;
+        } else if (!strcmp(argv[0], "ule")) {
+            ctx.operation = TPM2_EO_UNSIGNED_LE;
+        } else if (!strcmp(argv[0], "bs")) {
+            ctx.operation = TPM2_EO_BITSET;
+        } else if (!strcmp(argv[0], "bc")) {
+            ctx.operation = TPM2_EO_BITCLEAR;
+        } else {
+            LOG_ERR("Invalid Comparison-Operation specified.");
+            return false;
+        }
+
+    return true;
+}
+
+static bool process_input_operand_b(char *value) {
+
+    char *input_file = strcmp("-", value) ? value : NULL;
+    bool result;
+
+    if (input_file) {
+        result = files_get_file_size_path(value,
+                (long unsigned *) &ctx.operand_b.size);
+    }
+    if (input_file && !result) {
+        return false;
+    }
+
+    result = files_load_bytes_from_buffer_or_file_or_stdin(NULL, input_file,
+            (UINT16 *)&ctx.operand_b.size,
+            (unsigned char *)&ctx.operand_b.buffer);
+    if (!result) {
+        return false;
+    }
+
+    return true;
+}
+
+static bool process_input_offset(char *value) {
+
+    bool result = tpm2_util_string_to_uint16(value, &ctx.offset);
+    if (!result) {
+        LOG_ERR("Could not convert starting offset, got: \"%s\"", value);
+    }
+
+    return result;
+}
+
+static bool on_option(char key, char *value) {
+
+    bool result = true;
+
+    switch (key) {
+    case 'L':
+        ctx.policy_digest_path = value;
+        break;
+    case 'S':
+        ctx.session_path = value;
+        break;
+    case 'i':
+        result = process_input_operand_b(value);
+        goto on_option_out;
+    case 0:
+        result = process_input_offset(value);
+        goto on_option_out;
+    default:
+        return false;
+    }
+
+on_option_out:
+    return result;
+}
+
+bool tpm2_tool_onstart(tpm2_options **opts) {
+
+    const struct option topts[] = {
+        { "policy",    required_argument, NULL, 'L' },
+        { "session",   required_argument, NULL, 'S' },
+        { "input",     required_argument, NULL, 'i' },
+        { "offset",    required_argument, NULL,  0  },
+    };
+
+    *opts = tpm2_options_new("L:S:i:", ARRAY_LEN(topts), topts, on_option,
+            on_arg, 0);
+
+    return *opts != NULL;
+}
+
+bool is_input_option_args_valid(void) {
+
+    if (!ctx.session_path) {
+        LOG_ERR("Must specify -S session file.");
+        return false;
+    }
+
+    if (!ctx.operand_b.size) {
+        LOG_ERR("Data to compare is of size 0");
+        return false;
```

Comment:
is this really an error? Is it possible to compare against an array of 0 length? Or perhaps this should be a warning?

Reply:
If you error out and stop the tool this would be LOG_ERR